packages/web/hooks/window/use-localstorage-state.ts

```diff
@@ -26,16 +27,20 @@ export function useLocalStorageState<T>(
     }
   }, [key, setStoredValue]);
 
-  const setValue = (value: T) => {
-    try {
-      if (typeof window !== "undefined" && key) {
-        window.localStorage.setItem(key, JSON.stringify(value));
+  const setValue = useCallback(
+    (value: T) => {
+      try {
+        if (typeof window !== "undefined" && key) {
+          window.localStorage.setItem(key, JSON.stringify(value));
+        }
+
+        setStoredValue(value);
+      } catch (error) {
+        console.error(error);
       }
+    },
+    [key]
```

Comment:
is this the only dependency?

Reply:
Yes, this was also confirmed by Eslint React rule